wallet-next/src/wallet.rs

```diff
@@ -0,0 +1,144 @@
+use anyhow::Context;
+use penumbra_crypto::{
+    fmd,
+    keys::{
+        FullViewingKey, IncomingViewingKey, OutgoingViewingKey, SeedPhrase, SpendKey, SpendSeed,
+    },
+    Address, Note,
+};
+use serde::{Deserialize, Serialize};
+
+/// The contents of the wallet file that share a spend authority.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(from = "serde_helpers::WalletHelper")]
+#[serde(into = "serde_helpers::WalletHelper")]
+pub struct Wallet {
+    /// A list of human-readable labels for addresses.
+    ///
+    /// The label at index `i` is used for the address with `DiversifierIndex(i)`.
+    address_labels: Vec<String>,
+    spend_key: SpendKey,
```

Comment:
As a heads' up, we shouldn't have the wallet service hold onto a `SpendKey`, since it shouldn't have spending authority.  Instead, it should just hold a [`FullViewingKey`](https://rustdoc.penumbra.zone/main/penumbra_crypto/keys/struct.FullViewingKey.html), so that the wallet service isn't capable of spending funds.

Reply:
#671 will be helpful for this.